feat: Enable custom field search on POS

AfshanKhan opened the pull request:
### Use-case:
Consider Item has custom fields that have values in a language different than system language.

This feature enables users to search for those items with the values in those custom or standard fields in Point of Sale.

Added a table in POS Setting called **"POS Search Fields"**
![image](https://user-images.githubusercontent.com/33727827/115529985-fe572580-a2b0-11eb-8385-936dbf03de71.png)


The options for **"Field"** are the fields present in the Item master.
![image](https://user-images.githubusercontent.com/33727827/115676279-0d031280-a36d-11eb-95d7-a433b374d874.png)

After selection of any field that field's value will also be considered in the search for the item in Point of Sale.

### For example:
We have selected **"Description"**.
![image](https://user-images.githubusercontent.com/33727827/115530536-7de4f480-a2b1-11eb-9125-aa9f4b47f36e.png)

We added **"هواء"** in one of our Items description.
![image](https://user-images.githubusercontent.com/33727827/115530802-bdabdc00-a2b1-11eb-9dd5-84c59e5868e0.png)

### Now if we go to Point of Sale.
Without searching anything this is the screen.
![image](https://user-images.githubusercontent.com/33727827/115530960-dfa55e80-a2b1-11eb-8b03-e0809053427f.png)

After we search with keyword **"هواء"**
![image](https://user-images.githubusercontent.com/33727827/115531174-1aa79200-a2b2-11eb-942c-d153c3740c2d.png)

AfshanKhan committed:
feat: multi lingual item search in POS

feat: multi lingual item search in POS

AfshanKhan committed:
feat: multi lingual item search in POS

AfshanKhan committed:
Merge branch 'multi-lingual-item-search-pos' of https://github.com/AfshanKhan/erpnext into multi-lingual-item-search-pos

nextchamp-saqib commented on erpnext/selling/page/point_of_sale/point_of_sale.py:
```diff
@@ -139,8 +139,16 @@ def get_conditions(item_code, serial_no, batch_no, barcode):
+	cond = "("
+	cond += """item.name like {item_code}
+		or item.item_name like {item_code}""".format(item_code = frappe.db.escape('%' + item_code + '%'))
+	search_fields = frappe.get_all('POS Search Fields', fields = ['fieldname'])
```
It would be better if we moved this to a seperate function.

AfshanKhan replied:
Okay will do that

nextchamp-saqib commented on erpnext/selling/page/point_of_sale/point_of_sale.py:
```diff
@@ -139,8 +139,16 @@ def get_conditions(item_code, serial_no, batch_no, barcode):
+		or item.item_name like {item_code}""".format(item_code = frappe.db.escape('%' + item_code + '%'))
+	search_fields = frappe.get_all('POS Search Fields', fields = ['fieldname'])
+	if search_fields:
+		for field in search_fields:
```
We should probably verify if the field is not already used in the default condition. For eg. item_name

If such is the case, do not add the custom search condition. 

Or we could stop them from getting added to the child table itself by a python validation on POS settings.

AfshanKhan replied:
The better approach in my opinion is to not show those in the drop-down itself. Hence it won't be available for selection and solve our issue. What do you say?

AfshanKhan committed:
Merge branch 'develop' into multi-lingual-item-search-pos

AfshanKhan committed:
fix: limit the options for selection and broken down funtion for geting condition

AfshanKhan committed:
fix: limit the options for selection and broken down funtion for geting condition

fix: limit the options for selection and broken down funtion for geting condition

nextchamp-saqib commented on erpnext/selling/page/point_of_sale/point_of_sale.py:
```diff
@@ -81,7 +81,7 @@ def get_items(start, page_length, price_list, item_group, pos_profile, search_va
 			bin_join_selection=bin_join_selection,
 			bin_join_condition=bin_join_condition
-		), {'warehouse': warehouse}, as_dict=1)
+		), {'warehouse': warehouse}, as_dict=1, debug=True)
```
```suggestion
		), {'warehouse': warehouse}, as_dict=1)
```

AfshanKhan committed:
Merge branch 'multi-lingual-item-search-pos' of https://github.com/AfshanKhan/erpnext into multi-lingual-item-search-pos
Merge branch 'multi-lingual-item-search-pos' of https://github.com/AfshanKhan/erpnext into multi-lingual-item-search-pos

Merge branch 'multi-lingual-item-search-pos' of https://github.com/AfshanKhan/erpnext into multi-lingual-item-search-pos

AfshanKhan committed:
Merge branch 'develop' into multi-lingual-item-search-pos

nextchamp-saqib committed:
Merge branch 'develop' into multi-lingual-item-search-pos

nextchamp-saqib commented on erpnext/accounts/doctype/pos_settings/pos_settings.js:
```diff
@@ -21,6 +29,39 @@ frappe.ui.form.on('POS Settings', {
+			});
+
+			fields.unshift('');
+			frm.set_df_property('pos_search_fields', 'options', fields, cur_frm.docname, 'field');
```
```suggestion
			frm.fields_dict.pos_search_fields.grid.update_docfield_property('field', 'options', fields);
```

nextchamp-saqib replied:
Tried locally, the `update_docfield_property` works

nextchamp-saqib commented on erpnext/accounts/doctype/pos_settings/pos_settings.js:
```diff
@@ -21,6 +29,39 @@ frappe.ui.form.on('POS Settings', {
+	add_search_options: function(frm) {
+		frappe.model.with_doctype("Item", () => {
+			var fields = $.map(frappe.get_doc("DocType", "Item").fields, function(d) {
+				debugger;
```
```suggestion

```

nextchamp-saqib committed:
fix: remove debugger
